sdk/python/feast/infra/online_stores/redis.py

```diff
@@ -98,10 +98,16 @@ def update(
         partial: bool,
     ):
         """
-        We delete the keys in redis for tables/views being removed.
+        Look for join_keys (list of entities) that are not in use anymore
```

Comment:
I agree that this approach will work, but I'm questioning whether we want to preserve this data model. What do you think about switching to a model where the prefix contains not only the `join_keys`, but also the FV name - this way we can just actually delete the correct maps from Redis, instead of waiting for all FVs sharing the same entities to be deleted from the feature store?

Reply:
I agree that current model is not ideal. There are obvious trade offs. For example, keeping all features together is better for reading all those features if they are from different feature views. We can discuss this in more details on design meeting.
This PR's only goal is to prevent deleting data, that wasn't supposed to be deleted.

Reply:
> keeping all features together is better for reading all those features if they are from different feature views

I'd agree that this feels important. My gut says that online performance is much more important than code complexity in the offline/configuration paths.

There should be a way to have both I'd have thought - just needs some consideration.